File: algorithms/python/TwoSumIVInputisaBST/TwoSumIVInputisaBST.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findTarget(self, root: Optional[TreeNode], k: int) -> bool:
        ss = set()
        
        def dfs(node, ss, k):
            if not node: return False
            if k - node.val in ss: return True
            ss.add(node.val)
            if dfs(node.left, ss, k): return True
            if dfs(node.right, ss, k): return True
        
        return dfs(root, ss, k) == True

File: algorithms/js/TwoSumIVInputisaBST/TwoSumIVInputisaBST.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} k
 * @return {boolean}
 */
 var findTarget = function(root, k) {
    const map = {};
    
    const dfs = (node, map, k) => {
        if (!node) return false;
        if (map[k - node.val]) return true;        
        map[node.val] = 1;
        if (dfs(node.left, map, k)) return true;
        if (dfs(node.right, map, k)) return true;
    }
    return dfs(root, map, k) === true;
};